### DISCUSSION TITLE
Update document on elasticsearch

### DISCUSSION THREAD

**Discussion by blackrez** [vectordotdev/vector#21665] [Q&A] [answered]
Hello,

We are migrating from fluentd to vector and we are stuck to replace the update document with a same ID. FluentD supports upsert with Elasticsearch but this not the case of vector. Is there an another solution ?

**Comment by jszwedko** [comment]
Hi @blackrez ,

You should be able to set https://vector.dev/docs/reference/configuration/sinks/elasticsearch/#bulk.action to `index` to have Vector "upsert". The default is `create`.

**Reply by jszwedko** [reply]
Actually, the default is `index`. Are you finding it not to be upserting in your case? You may need to set https://vector.dev/docs/reference/configuration/sinks/elasticsearch/#id_key

**Reply by blackrez** [reply]
It will not have the same comportement than we expected. Upsert will update/merge documents with the same ID, index will keep the last document.

**Reply by jszwedko** [reply]
Ah, I see, I was thinking you just wanted to update the doc, but you want to merge. I think this is a feature request being tracked here: https://github.com/vectordotdev/vector/issues/17852

**Reply by blackrez** [reply]
This is clearly the feature we needed. Do you think this is possible to add this feature in a near future ?

**Reply by jszwedko** [reply]
We are open to contributions but otherwise we can't promise a timeline. I can say that feature requests are reviewed and prioritized based on expected impact and demand.

**Reply by blackrez** [reply]
For the impact, I can say it will help a lot of customers using Elasticsearch with fluentd or logstash to migrate for vector 😉 .

**Reply by blackrez** [reply]
> We are open to contributions but otherwise we can't promise a timeline. I can say that feature requests are reviewed and prioritized based on expected impact and demand.

Hello, my team has agreed to work on it. We are seeking fo good advices for the implementation, so if you have some I will be happy to apply them ?

**Reply by pront** [reply, marked as answer]
Replied here: https://github.com/vectordotdev/vector/issues/17852#issuecomment-2471125679
